pkg/execution/executor/validate.go

```diff
@@ -78,26 +77,18 @@ func (r *runValidator) checkStepLimit(ctx context.Context) error {
 		return fmt.Errorf("%d is greater than the absolute step limit of %d", limit, consts.AbsoluteMaxStepLimit)
 	}
 
-	if limit > 0 && len(r.s.Actions()) >= limit {
-		// Update this function's state to overflowed, if running.
-		if r.md.Status == enums.RunStatusRunning {
-			// XXX: Update error to failed, set error message
-			if err := r.e.sm.SetStatus(ctx, r.md.Identifier, enums.RunStatusFailed); err != nil {
-				return err
-			}
-
-			// Create a new driver response to map as the function finished error.
-			resp := state.DriverResponse{}
-			resp.SetError(state.ErrFunctionOverflowed)
-			resp.SetFinal()
+	if limit > 0 && r.md.Metrics.StepCount >= limit {
+		// Create a new driver response to map as the function finished error.
+		resp := state.DriverResponse{}
+		resp.SetError(state.ErrFunctionOverflowed)
+		resp.SetFinal()
```

Comment:
should this also run `SaveStep`?
otherwise, how will the status for the function be set if it's going to return early?

Reply:
Ah I see. I was asking saving for history, not state. if that's handled, I think it should be fine.